controllers/sdkutil/spec_comparison.go

```diff
@@ -39,7 +39,7 @@ type Comparison struct {
 // Determine if the given TrainingJobSpec matches the DescribeTrainingJobOutput. This converts the description to a TrainingJobSpec,
 // then selectively compares fields.
 func TrainingJobSpecMatchesDescription(description sagemaker.DescribeTrainingJobOutput, spec trainingjobv1.TrainingJobSpec) Comparison {
-	remoteSpec := CreateTrainingJobSpecFromDescription(description)
+	remoteSpec, _ := CreateTrainingJobSpecFromDescription(description)
```

Comment:
Every field specified in the spec will be different from that specified in the description (no spec will match any description). This `TrainingJobSpecMatchesDescription` doesn't seem to be used anywhere within the codebase anymore. I suppose it used to work with TrainingJob to reconcile the two, but no longer is necessary?

Reply:
Should I simply delete this method if it is never being called? Why TODO or simply comment it?

In regards to my previous comment, what I meant to say was that if the method returns a (nil, error) tuple, the comparison returned by this method will show that no field in the description (the nil object) matches the training job.

Reply:
> Should I simply delete this method if it is never being called? Why TODO or simply comment it?

I like to think holistically when making changes to reduce rough edges when the change is implemented. If we have a TODO, when we revisit this in the JSON refactor, we can make sure to remove this completely. I'm not opposed to removing since it is dead code, we just need to be vigilant in JSON refactor to not leave any rough edges.